content/en/docs/how-tos/staged-update.md

```diff
@@ -247,123 +274,180 @@ status:
+        type: ApprovalRequestCreated
+      type: Approval
+    clusters:
+    - clusterName: member3
+    - clusterName: member1
     conditions:
     - lastTransitionTime: ...
       message: ""
-      observedGeneration: 1
-      reason: StageUpdatingStarted
-      status: "True" # stage canary is still executing
+      observedGeneration: 2
+      reason: StageUpdatingWaiting
+      status: "False"
       type: Progressing
     stageName: canary
-    startTime: ...
 ```
-Wait a little bit more, and we can see stage `canary` finishes cluster update and is waiting for the Approval task.
-We can check the `ClusterApprovalRequest` generated and not approved yet:
+After stage `staging` completes, the canary stage requires approval **before** it starts (due to beforeStageTasks). Check for the before-stage approval request:
+```bash
+kubectl get clusterapprovalrequest -A
+NAME                        UPDATE-RUN    STAGE    APPROVED   AGE
+example-run-before-canary   example-run   canary              6m55s
+```
+
+Approve the before-stage task to allow canary stage to start:
 ```bash
-kubectl get clusterapprovalrequest
-NAME                 UPDATE-RUN    STAGE    APPROVED   APPROVALACCEPTED   AGE
-example-run-canary   example-run   canary                                 2m2s
+kubectl patch clusterapprovalrequests example-run-before-canary --type='merge' \
+  -p '{"status":{"conditions":[{"type":"Approved","status":"True","reason":"approved","message":"approved","lastTransitionTime":"'$(date -u +"%Y-%m-%dT%H:%M:%SZ")'","observedGeneration":2}]}}' \
+  --subresource=status
 ```
-We can approve the `ClusterApprovalRequest` by patching its status:
+
+Once approved, the canary stage begins updating clusters. With `maxConcurrency: 2`, it updates up to 2 clusters concurrently.
+
+Wait for the canary stage to finish cluster updates. It will then wait for the after-stage Approval task:
+```bash
+kubectl get clusterapprovalrequest -A
+NAME                        UPDATE-RUN    STAGE    APPROVED   AGE
+example-run-after-canary    example-run   canary              3s
+example-run-before-canary   example-run   canary   True       15m
+```
+
+> Note: Observed generation in the Approvaed condition should match the generation of the updateRun object.
+
+Approve the after-stage task to complete the rollout:
 ```bash
-kubectl patch clusterapprovalrequests example-run-canary --type=merge -p {"status":{"conditions":[{"type":"Approved","status":"True","reason":"lgtm","message":"lgtm","lastTransitionTime":"'$(date -u +%Y-%m-%dT%H:%M:%SZ)'","observedGeneration":1}]}} --subresource=status
-clusterapprovalrequest.placement.kubernetes-fleet.io/example-run-canary patched
+kubectl patch clusterapprovalrequests example-run-after-canary --type='merge' \
+  -p '{"status":{"conditions":[{"type":"Approved","status":"True","reason":"approved","message":"approved","lastTransitionTime":"'$(date -u +"%Y-%m-%dT%H:%M:%SZ")'","observedGeneration":2}]}}' \
```

Comment:
I'm curious: why would the generation get changed in approvalRequests? We only create the approvalRequest obj without any updates?